### PR TITLE
Sort the speakers the same as the sessions

### PR DESCRIPTION
**Pull request by adrianbanks**
This sorts them alphabetically by name, with the current speaker first.

### CODE REVIEW

File: DDDEastAnglia/Controllers/SpeakerController.cs [hunk 1]
```diff
@@ -27,6 +28,7 @@ public virtual ActionResult Index()
                 }
             }
 
+            speakers.Sort(new SpeakerDisplayModelComparer());
```
Comment (alastairs) [inline DDDEastAnglia/Controllers/SpeakerController.cs:30]: Out of curiosity, is this any better or worse than using `OrderBy()`?

File: DDDEastAnglia/Controllers/SpeakerController.cs [hunk 2]
```diff
@@ -66,4 +68,22 @@ private SpeakerDisplayModel CreateDisplayModel(UserProfile userProfile, IEnumera
                 };
         }
     }
+
+    public class SpeakerDisplayModelComparer : IComparer<SpeakerDisplayModel>
```
Comment (alastairs) [inline DDDEastAnglia/Controllers/SpeakerController.cs]: Should be in a separate file. Or, as it's an implementation detail, a private nested class.

File: DDDEastAnglia/Controllers/SpeakerController.cs (hunk 1 above)
Comment (adrianbanks) [inline DDDEastAnglia/Controllers/SpeakerController.cs:30]: Yes, you can't do custom (complex) sorting using `OrderBy()`. I want the current user's profile (and sessions in the other controller) to be at the top.

Comment (alastairs) [inline DDDEastAnglia/Controllers/SpeakerController.cs:30]: Hmm, I thought you could, but perhaps that's only in Linq2Sql/Entity Framework.

**Pull request merged**